Shared-Library/hello-world.hpp

```diff
@@ -0,0 +1,5 @@
+#pragma once
+
+string say_hello();
+
+int main();
```

Comment:
shared libraries don't have main functions, also there's no point in forward delcaring the main function since it defines the API to an executable and not a library.